src/mono/wasm/runtime/startup.ts

```diff
@@ -53,7 +53,7 @@ export function configure_emscripten_startup(module: DotnetModule, exportedAPI:
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     const userOnRuntimeInitialized: () => void = module.onRuntimeInitialized ? module.onRuntimeInitialized : () => { };
     // when assets don't contain DLLs it means this is Blazor or another custom startup
-    isCustomStartup = !module.configSrc && (!module.config || !module.config.assets || module.config.assets.findIndex(a => a.behavior === "assembly") != -1); // like blazor
+    isCustomStartup = !module.configSrc && (!module.config || !module.config.assets || module.config.assets.findIndex(a => a.behavior === "assembly") == -1); // like blazor
```

Comment:
Why the change?

Reply:
it's a bug. 

if there are assemblies in the list of assets, it means that:
a) it's not blazor
b) it's blazor fully converted to use full, non-custom startup sequence (net8)

if there are NO assemblies in the list of assets, it means that:
c) it's blazor and it still needs custom startup sequence (net7)